### DISCUSSION TITLE
Issue with python-escpos and pyBluez integration for Bluetooth ESCPOS printers

### DISCUSSION THREAD

**Discussion by ismaelit** [python-escpos/python-escpos#517] [Ideas]
Dear GitHub Team,

I hope this message finds you well. I am writing to discuss an issue I encountered while attempting to integrate python-escpos with pyBluez for printing on Bluetooth ESCPOS printers. While I was able to achieve success using `printer.Dummy()` and sending everything to the RFCOMM Bluetooth socket, I encountered a problem with long messages being truncated and not printed properly. (Using USB, the entire message is printed normally)

I have made several attempts to reduce the size of Bluetooth packets, but unfortunately, I have not been successful. On the other hand, the pyEscpos library has been successful in utilizing Bluetooth with ESCPOS printers. However, I would prefer to use python-escpos, as I have already used it in other projects involving network and USB printing, and easier to deal with encodings and character tables.

I wanted to inquire whether there are any plans to integrate python-escpos with Bluetooth printers via pyBluez. Alternatively, if there is another recommended approach or workaround for this issue, I would greatly appreciate any guidance or suggestions.

Best regards,

Ismael

**Comment by tegarnugroho** [comment]
Hi @ismaelit  after about 1 month you opened this discussion, what is your decision? do you still use python-espos and combine with pybluez or do you migrate to pyEscpos ?

**Reply by ismaelit** [reply]
Since yesterday I'm trying to change the approach: now using a raspberry pi that has bluetooth classic (my printer is a portable EPSON TM-P80 Plus, and only has bluetooth classic), and I'm using pythons serial mode with rfcomm simulating a serial port on the system layer (raspbian) and no longer in Python. So far I am able to print long messages via bluetooth.

Things to do/fix right now:
1- After the first bluetooth connection to the raspberry pi, on the first printing, I get junk codes on the first line;
2- Dockerize an application and try to use the system's rfcomm port;
3- Automate the connection with the bluetooth printer, which I am currently doing manually: sudo rfcomm connect /dev/rfcomm0 AA:BB:CC:DD:EE:FF 1 &

**Reply by ismaelit** [reply]
fix 1: just insert the printer reset as first command before print: 
p._raw(b'\x1b' + b'@')
